pkg/reconciler/sugar/trigger/trigger.go

```diff
@@ -20,35 +20,61 @@ import (
 	"context"
 	"fmt"
 
+	"go.uber.org/zap"
 	corev1 "k8s.io/api/core/v1"
 	k8serrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/dynamic"
+
+	pkgduck "knative.dev/pkg/apis/duck"
+	duckv1 "knative.dev/pkg/apis/duck/v1"
+	"knative.dev/pkg/logging"
+	"knative.dev/pkg/reconciler"
+
+	eventingv1 "knative.dev/eventing/pkg/apis/eventing/v1"
 	"knative.dev/eventing/pkg/apis/eventing/v1beta1"
 	clientset "knative.dev/eventing/pkg/client/clientset/versioned"
 	triggerreconciler "knative.dev/eventing/pkg/client/injection/reconciler/eventing/v1beta1/trigger"
 	listers "knative.dev/eventing/pkg/client/listers/eventing/v1beta1"
+	"knative.dev/eventing/pkg/duck"
 	"knative.dev/eventing/pkg/reconciler/sugar"
 	"knative.dev/eventing/pkg/reconciler/sugar/resources"
-	"knative.dev/pkg/logging"
-	"knative.dev/pkg/reconciler"
 )
 
 const (
 	// Name of the corev1.Events emitted from the reconciliation process.
 	brokerCreated = "BrokerCreated"
+
+	// Name of the CE overrides extension uniquely identifying a source
+	sourceIdExtension = "knsourcetrigger"
 )
 
 type Reconciler struct {
 	// eventingClientSet allows us to configure Eventing objects
 	eventingClientSet clientset.Interface
 	brokerLister      listers.BrokerLister
 	isEnabled         sugar.LabelFilterFn
+
+	// Dynamic tracker to track Sources. In particular, it tracks the dependency between Triggers and Sources.
+	sourceTracker    duck.ListableTracker
+	dynamicClientSet dynamic.Interface
 }
 
 // Check that our Reconciler implements triggerreconciler.Interface
 var _ triggerreconciler.Interface = (*Reconciler)(nil)
 
 func (r *Reconciler) ReconcileKind(ctx context.Context, t *v1beta1.Trigger) reconciler.Event {
+	event := r.reconcileInjection(ctx, t)
+	if event != nil {
+		return event
+	}
+
+	return r.reconcileDependency(ctx, t)
+}
+
+func (r *Reconciler) reconcileInjection(ctx context.Context, t *v1beta1.Trigger) reconciler.Event {
 	if !r.isEnabled(t.GetAnnotations()) {
 		logging.FromContext(ctx).Debug("Injection for Trigger not enabled.")
```

Comment:
This by default checks for the injection annotation, right? We don't need to specify key: `InjectionAnnotation` now that we have more annotations?

Reply:
no need. The key is encapsulated in isEnabled.

Reply:
ok, great.